enumerable.rb

```diff
@@ -0,0 +1,176 @@
+module Enumerable
+  # 1. each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < size
+      if is_a?(Array)
+        yield(self[index])
+      elsif is_a?(Range)
+        yield(to_a[index])
+      end
+      index += 1
+    end
+    yield(keys.zip(values)) if is_a?(Hash)
+  end
+
```

Comment:
enumerables #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self